pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,319 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository that implements an in-memory repository
+based on dictionaries
+
+For the module-level documentation, see mocksupport.rst.
+"""
+
+import six
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+from pywbem._utils import _format
+from ._baserepository import BaseObjStore, BaseRepository
+
```

Comment:
Do we want InMemoryObjStore to be part of the public API? If so, define an `__all__` list containing it.

Reply:
Not sure at this point but created __all__ with both classes

Reply:
We need to document InMemoryObjStore, because we return it and the caller needs to understand how it is being used. That works only if it is part of the pywbem_mock (Python) namespace. So I think it should be added to `__all__`.

Reply:
has been DONE.